examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -236,3 +238,27 @@ void BdxOtaSender::Reset()
     mNumBytesSent = 0;
     memset(mFileDesignator, 0, chip::bdx::kMaxFileDesignatorLen);
 }
+
+CHIP_ERROR BdxOtaSender::GracefullyCloseTransfer()
+{
+    using namespace chip::Protocols::SecureChannel;
+
+    StatusReport statusReport(GeneralStatusCode::kFailure, Id, kProtocolCodeGeneralFailure);
```

Comment:
There's already a function in the BdxTransferSession code called AbortTransfer that will build a queue a status message for you.

It's just a matter of calling this at the right time, and ensuring that the report is sent out at the proper time. So I think this might be a matter of calling AbortTransfer, and then adding something into the TransferSession::HandleMessageReceived function to send out pending status reports before handling the BdxMessage (to ensure the message you're sending out isn't overwritten by an incorrect state message)